client/components/domains/domain-registration-suggestion/index.jsx

```diff
@@ -29,10 +31,25 @@ const DomainRegistrationSuggestion = React.createClass( {
 			isAdded = hasDomainInCart( this.props.cart, suggestion.domain_name ),
 			buttonClasses,
 			buttonContent,
-			domainFlag = null;
+			domainFlags = [],
+			translate = this.props.translate;
 
 		if ( suggestion.domain_name ) {
-			domainFlag = <DomainSuggestionFlag domain={ suggestion.domain_name }/>;
+			const newTLDs = [];
+
+			if ( newTLDs.some(
+					( tld ) => endsWith( suggestion.domain_name, tld ) && suggestion.domain_name.substring( 0, suggestion.domain_name.length - ( tld.length + 1 ) ).indexOf( '.' ) === -1
+				) ) {
+				domainFlags.push( <DomainSuggestionFlag content={ translate( 'New', { context: 'Domain suggestion flag' } ) }/> );
+			}
+		}
+
+		if ( suggestion.isRecommended ) {
+			domainFlags.push( <DomainSuggestionFlag content={ translate( 'Recommended', { context: 'Domain suggestion flag' } ) }/> );
```

Comment:
Also, the comment should have suggested "Recommended" without a context as its first alternative btw - I'll check to see why it didn't.